.github/workflows/pr_closed.yaml

```
@@ -0,0 +1,92 @@
+        # https://github.com/docker/hub-tool/pull/198
+        # https://github.com/docker/hub-tool/issues/58#issuecomment-1006058934
+        run: |
+          #curl -sO $(curl -sL https://api.github.com/repos/docker/hub-tool/releases/latest | jq -r '.assets[].browser_download_url' | grep linux-amd64.tar.gz) | tar -xz  -C /usr/bin
+          git clone https://github.com/docker/hub-tool.git
+          cd hub-tool
+          git checkout c9cb0ac822ad431a252d462d80ab0dcc21d49e6f
+          cat << "EOF" > fixes.patch
+          diff -ur hub-tool/internal/commands/tag/rm.go hub-tool-envlogin/internal/commands/tag/rm.go
+          --- hub-tool/internal/commands/tag/rm.go	2022-01-05 13:04:09.000000000 -0800
+          +++ hub-tool-envlogin/internal/commands/tag/rm.go	2022-01-05 13:04:58.000000000 -0800
+          @@ -96,7 +96,12 @@
+           	}
+          
+           	if err := hubClient.RemoveTag(reference.FamiliarName(ref), ref.Tag()); err != nil {
+          -		return err
+          +		if strings.Contains(err.Error(), "404 NOT FOUND") {
+          +			fmt.Fprintln(streams.Out(), "Not Found", image)
+          +			return nil
+          +		} else {
+          +			return err
+          +		}
+           	}
+           	fmt.Fprintln(streams.Out(), "Deleted", image)
+           	return nil
+          diff -ur hub-tool/internal/login/login.go hub-tool-envlogin/internal/login/login.go
+          --- hub-tool/internal/login/login.go	2022-01-05 13:04:09.000000000 -0800
+          +++ hub-tool-envlogin/internal/login/login.go	2022-01-05 10:18:07.000000000 -0800
+          @@ -40,14 +40,20 @@
+           func RunLogin(ctx context.Context, streams command.Streams, hubClient *hub.Client, store credentials.Store, candidateUsername string) error {
+           	username := candidateUsername
+           	if username == "" {
+          +		username = os.Getenv("DOCKER_USERNAME")
+          +	}
+          +	if username == "" {
+           		var err error
+           		if username, err = readClearText(ctx, streams, "Username: "); err != nil {
+           			return err
+           		}
+           	}
+          -	password, err := readPassword(streams)
+          -	if err != nil {
+          -		return err
+          +	password := os.Getenv("DOCKER_PASSWORD")
+          +	if password == "" {
+          +		var err error
+          +		if password, err = readPassword(streams); err != nil {
+          +			return err
+          +		}
+           	}
+          
+           	token, refreshToken, err := Login(ctx, streams, hubClient, username, password)
+          EOF
+          patch -s -p1 < ./fixes.patch
+          make
+          sudo mv bin/hub-tool /usr/bin/hub-tool
+          chmod a+rx /usr/bin/hub-tool
+      - name: Remove PR container image via hub-tool
+        run: |
+          /usr/bin/hub-tool login
+          /usr/bin/hub-tool tag rm -f superorbital/cludo:development.git-PR-${PR_NUM}
```

I agree with [your thoughts on how `-f` should work](https://github.com/docker/hub-tool/issues/58#issuecomment-1006058934), but in the meantime, couldn't we append `|| true` to get the same behavior?  This would allow us to skip the whole patch/build step above, right?

```suggestion
          /usr/bin/hub-tool tag rm -f superorbital/cludo:development.git-PR-${PR_NUM} || true
```

---

Not Really.

There are 2 patches there. One so that we can login non-interactively, and one so that `-f` does not report an error when the tag does not exist. We also would really need more logic beyond `| true` as this would also mask real permissions issues.

We likely need to simply fork it for now, but I started here. (see: https://github.com/superorbital/cludo/issues/67)